lib/src/models/data_bundle.dart

```diff
@@ -40,35 +41,61 @@ class DataBundle {
 
   static Future<DataBundle> fromHandles({
     required List<DataItemHandle> handles,
+    List<DataItem> items = const [],
   }) async {
-    final headers = Uint8List(64 * handles.length * 2);
+    final noOfDataItems = (handles.length * 2) + items.length;
+    final headers = Uint8List(64 * noOfDataItems);
     // Use precalculated buffers if provided to
     var dataItemIndex = 0;
     final binaries = BytesBuilder();
-    await Future.wait(handles.map((handle) async {
-      // Sign DataItem
-      final dataItems = await handle.createDataItemsFromFileHandle();
-      assert(dataItems.length == 2);
-      for (var dataItem in dataItems) {
-        final id = decodeBase64ToBytes(dataItem.id);
-        // Create header array
-        final header = Uint8List(64);
-        final raw = await dataItem.asBinary();
-        dataItem.data = Uint8List(0);
-        // Set offset
-        header.setAll(0, longTo32ByteArray(raw.length));
-        // Set id
-        header.setAll(32, id);
-        // Add header to array of headers
-        headers.setAll(64 * dataItemIndex, header);
-        dataItemIndex++;
-        // Convert to array for flattening
-        binaries.add(raw.takeBytes());
-      }
-    }));
-
+    await Future.wait(
+      handles.map(
+        (handle) async {
+          // Sign DataItem
+          final dataItems = await handle.createDataItemsFromFileHandle();
+          assert(dataItems.length == 2);
+          for (var dataItem in dataItems) {
+            final id = decodeBase64ToBytes(dataItem.id);
+            // Create header array
+            final header = Uint8List(64);
+            final raw = await dataItem.asBinary();
+            dataItem.data = Uint8List(0);
+            // Set offset
+            header.setAll(0, longTo32ByteArray(raw.length));
+            // Set id
+            header.setAll(32, id);
+            // Add header to array of headers
+            headers.setAll(64 * dataItemIndex, header);
+            dataItemIndex++;
+            // Convert to array for flattening
+            binaries.add(raw.takeBytes());
+          }
+        },
+      ),
+    );
+    await Future.wait(
```

Comment:
The logic here is the same of the above. What do you think about split it on another function?

Reply:
Looks like you're reviewing the changes from the first commit. This was updated.

Reply:
Yes! Sorry, Javed.